### PR TITLE
chore: lint and format on pre-commit and ci

### LINKED ISSUE #210: Check formatting and eslint in CI + review pre-commit hook
There have been a few PRs that had style modifications in files or Readme examples of code that were no actual changes as part of the PR that was being opened. 

We are not enforcing our defined styling enough (I think it's only run in a `pre-commit` hook) which can be skipped. This ticket is to

- [ ] Review the pre-commit hook to ensure it is actually working, and (except for manual skip by the dev) that it always runs when committing new code

- [ ] Update our CI process to verify the code to be merged is following the eslint and prettier rules, preventing new code from being merged if it does not follow our style guide

### PR DESCRIPTION
**Pull request by gndelia**
closes #210 

I also updated 4 dev dependencies

- lint-staged
- husky
- is-ci-cli (not required for the PR, but it's not a breaking change and given I was updating the CI conf...)
- prettier => in this case I did not go to the last up-to-date version, but to the latest stable without breaking changes - in this case 1.19.1. That's because in 1.19 they included support for parsing optional chaining `(foo?.baz)` which we use in TS files.
I could update prettier to the latest but I thought it'd be better to do it in a separated PR as I think they changed some defaults and it may do the PR larger than it's required.

The breaking changes of these ☝️  is just dropping old node support, which is not required for linting  our code

it's the same as #215 but targeting v4 instead of master 😅 

the changes of this PR in ts files are formatting stuff (automatically applied by prettier) or eslint fixes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,12 @@ before_script:
 
 jobs:
   include:
+    - stage: test
```
Comment (Belco90) [inline .travis.yml:20]: This one should be validation stage right?

**Comment by Belco90** [comment]
Agree on upgrading prettier in a different task.

File: .travis.yml (hunk 1 above)
Comment (gndelia) [reply .travis.yml:20]: I thought the stages where only test, build and deploy, that's why I went with this one. If validation is a valid one, it makes sense. I don't like travis ci docs 😛

Comment (gndelia) [reply .travis.yml]: Updated in https://github.com/testing-library/eslint-plugin-testing-library/pull/216/commits/b126b8d4b41fa86a7280305f13331cae9c15b7b2

Comment (Belco90) [reply .travis.yml]: I'd say stages can have any name: https://config.travis-ci.com/ref/stages the only thing is to be consistent around the CI config.

**Comment by Belco90** [comment]
Now we have corresponding stage for Validation ✨ 
CleanShot 2020-08-17 at 10 55 03

**Comment by gndelia** [comment]
> Now we have corresponding stage for Validation ✨
> CleanShot 2020-08-17 at 10 55 03

🚀

**Pull request merged**